### DISCUSSION TITLE
FileSurfer/WebSurfer

### DISCUSSION THREAD

**Discussion by asheeshgarg** [microsoft/autogen#4738] [General]
We have moved the file surfer and websurfer to ext package.  Also I see similar component in magentic one. will magentic ones will be removed eventaully? Do we have some notebook or test case for the file surfer for either packages. I see it always fails with invalid path error in magentic one?

**Comment by afourney** [comment]
Yes, both agents will be distributed as part of ext in the future, and built in AgentChat. It's a work in progress currently, and yes we'll add tests etc. At present, testing has consisted of re-running the GAIA benchmark (validation set of 165 questions).

As for the autogen_magentic_one folder, we will likely keep it around in a branch, or similar, because it is necessary to replicate the paper. But future development and bug fixes will happen in the ext package.

Finally, as for the path error, can you please provide a trace or sample code? I don't doubt there is a problem, but it isn't replicating in the narrow test harness I've been using.

**Reply by asheeshgarg** [reply]
@afourney so future we will move the magnetic one orchestrator as well to core? For AgentChat to use the ledger based communication and planning?

this is the error I get when ask to open a file. I am assuming it will open a file for me which is store in local. Don't see any test cases or notebook so want to make sure I am not assuming any thing wrong here
  File "I:\work\code\autogen\python\.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\requests_markdown_browser.py", line 287, in _fetch_page
    res = self._markdown_converter.convert_local(download_path)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "I:\work\code\autogen\python\.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\mdconvert.py", line 862, in convert_local
    self._append_ext(extensions, self._guess_ext_magic(path))
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "I:\work\code\autogen\python\.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\mdconvert.py", line 1015, in _guess_ext_magic
c
    guesses = puremagic.magic_file(path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "I:\work\code\autogen\python\.venv\Lib\site-packages\puremagic\main.py", line 316, in magic_file
", line 316, in magic_file
    head, foot = _file_details(filename)
                 ^^^^^^^^^^^^^^^^^^^^^^^
  File "I:\work\code\autogen\python\.venv\Lib\site-packages\puremagic\main.py", line 211, in _file_details
    with open(filename, "rb") as fin:
         ^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument: '\\h:\\work\\documents\\msds\\nlp\\fp.pdf'

**Reply by NaveendasAlgo** [reply]
> @afourney so future we will move the magnetic one orchestrator as well to core? For AgentChat to use the ledger based communication and planning?
> 
> this is the error I get when ask to open a file. I am assuming it will open a file for me which is store in local. Don't see any test cases or notebook so want to make sure I am not assuming any thing wrong here File "I:\work\code\autogen\python.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\requests_markdown_browser.py", line 287, in _fetch_page res = self._markdown_converter.convert_local(download_path) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "I:\work\code\autogen\python.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\mdconvert.py", line 862, in convert_local self._append_ext(extensions, self._guess_ext_magic(path)) ^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "I:\work\code\autogen\python.venv\Lib\site-packages\autogen_magentic_one\markdown_browser\mdconvert.py", line 1015, in _guess_ext_magic c guesses = puremagic.magic_file(path) ^^^^^^^^^^^^^^^^^^^^^^^^^^ File "I:\work\code\autogen\python.venv\Lib\site-packages\puremagic\main.py", line 316, in magic_file ", line 316, in magic_file head, foot = _file_details(filename) ^^^^^^^^^^^^^^^^^^^^^^^ File "I:\work\code\autogen\python.venv\Lib\site-packages\puremagic\main.py", line 211, in _file_details with open(filename, "rb") as fin: ^^^^^^^^^^^^^^^^^^^^ OSError: [Errno 22] Invalid argument: '\h:\work\documents\msds\nlp\fp.pdf'

Facing similar iisue
